go/store/types/tuple.go

```diff
@@ -497,6 +497,32 @@ func (t Tuple) fieldsToMap() (map[Value]Value, error) {
 	return valMap, nil
 }
 
+func (t Tuple) StartsWith(nbf *NomsBinFormat, otherTuple Tuple) (bool, error) {
```

Comment:
Does this work:

```
func (t Tuple) StartsWith(otherTuple Tuple) bool {
	return bytes.HasPrefix(t.buff, otherTuple.buff)
}
```

Reply:
Should add tests to tuple_test.go for StartsWith

Reply:
After some more thought, `HasPrefix` should actually work just fine. So I chose to not use it originally as I thought that you could potentially have a byte or string array as a Value in the Tuple, that could be a byte-wise match up to some point. For example:
```go
tupleA := NewTuple(nbf, types.Uint(1), types.InlineBlob{2, 3}, types.Uint(2), ...)
tupleB := NewTuple(nbf, types.Uint(1), types.InlineBlob{2, 3, ...})
```
So I thought that `tupleB` could have some end up having some value that would be equivalent to having `types.Uint(2)` encoded after as far as the raw bytes go, but after thinking again it wouldn't hold up, since all variable-length types also encode their length, so they would fail the prefix check since the length would be different. My worry isn't a real issue in practice.

Reply:
`bytes.HasPrefix` should also be a good deal faster too so it's a win for everyone.